packages/chatbot-server-mongodb-public/src/tracing/routesUpdateTraceHandlers.ts

```diff
@@ -128,52 +128,57 @@ ${tracingData.assistantMessage?.content}
 
     // Send Segment events
     try {
-      if (segmentTrackUserSentMessage) {
+      const userMessage = tracingData.userMessage;
+      const { userId, anonymousId } = getSegmentIds(userMessage);
+      const hasSegmentId = !!userId || !!anonymousId;
```

Comment:
Can you add this `hasSegmentId` check to the other Segment trace functions in this file?

Should exist for:
- AddMessage
- RateMessage
- CommentMessage